src/contentscript/contexts/mutable-web-context/mutable-web-provider.tsx

```diff
@@ -33,6 +33,10 @@ const MutableWebProvider: FC<Props> = ({ children, engine }) => {
     [mutations, selectedMutationId]
   )
 
+  useEffect(() => {
+    setSelectedMutationId(selectedMutation?.id ?? null)
+  }, [selectedMutation])
```

Comment:
It's not necessary and  can cause looping calculations because
* `selectedMutationId` is calculated from `selectedMutation`
* `selectedMutation` is calculated from `selectedMutationId`